network/syncer/cursors_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package syncer
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethersphere/swarm/network/simulation"
+)
+
+var (
+	bucketKeyFileStore = simulation.BucketKey("filestore")
+	bucketKeyBinIndex  = simulation.BucketKey("bin-indexes")
+	bucketKeySyncer    = simulation.BucketKey("syncer")
+)
+
+// TestNodesExchangeCorrectBinIndexes tests that two nodes exchange the correct cursors for all streams
+// it tests that all streams are exchanged
+func TestNodesExchangeCorrectBinIndexes(t *testing.T) {
+	nodeCount := 2
+
+	// create a standard sim
+	sim := simulation.New(map[string]simulation.ServiceFunc{
+		"bzz-sync": newBzzSyncWithLocalstoreDataInsertion,
+	})
+	defer sim.Close()
+
+	// create context for simulation run
```

Comment:
Redundant comment, and `ctx` declaration should be right before the `sim.Run`. The comment above `sim.Run` is also redundant.